cmake/compiler/gcc/target.cmake

```diff
@@ -6,10 +6,33 @@ set_ifndef(C++ g++)
 # GCC-based toolchains
 
 if("${SPARSE}" STREQUAL "y")
-  find_program(CMAKE_C_COMPILER cgcc)
-  find_program(REAL_CC ${CROSS_COMPILE}${CC} PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
-  set(ENV{REAL_CC} ${REAL_CC})
-else()
+  # No search PATHS because we need more than cgcc in the default PATH
+  find_program(CMAKE_C_COMPILER cgcc REQUIRED)
+  message(STATUS "sparse configuration: cgcc=${CMAKE_C_COMPILER}")
+
+  find_program(SPARSE_REAL_COMPILER ${CROSS_COMPILE}${CC} PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH REQUIRED)
+
+  # We know what REAL_CC _must_ be, so why do we ask the user to define it?  Because CMake
+  # cannot set (evil) build time env variables at configure time:
+  # https://gitlab.kitware.com/cmake/community/-/wikis/FAQ#how-can-i-get-or-set-environment-variables
+  # As of Sep. 2022, sparse/cgcc has unfortunately no --real-cc option as it should.
+  #
+  # In theory we could define REAL_CC ourselves here and fix the configure-time
+  # --print-file-name below (and maybe others). Then ask the user to define REAL_CC later
+  # at _build_ time. But in practice who would define environment variables at build time
+  # _only_?  So best to fail early and clearly here.
+  if ("$ENV{REAL_CC}" STREQUAL "")
+    message(FATAL_ERROR "sparse's REAL_CC is undefined, it must be set to: ${SPARSE_REAL_COMPILER}")
+  else() # check ENV{REAL_CC}
+    file(REAL_PATH "${SPARSE_REAL_COMPILER}" real_compiler_rp)
+    file(REAL_PATH "$ENV{REAL_CC}" env_real_cc_rp)
+    cmake_path(COMPARE "${real_compiler_rp}" EQUAL "${env_real_cc_rp}" expected_env_REAL_CC)
+    if(NOT expected_env_REAL_CC)
+      message(FATAL_ERROR "Unexpected value: REAL_CC=$ENV{REAL_CC}, should be: ${SPARSE_REAL_COMPILER}")
```

Comment:
User is not informed from where `REAL_CC` was defined.

One could believe doing a `set(REAL_CC <path>)` in application CMakeLists.txt or a `-DREAL_CC=<path>` on invocation will fix the problem, which is not the case.